pkg/services/provisioning/alerting/rules_provisioner.go

```diff
@@ -85,9 +85,9 @@ func (prov *defaultAlertRuleProvisioner) provisionRule(
 		return err
 	} else if err != nil {
 		prov.logger.Debug("creating rule", "uid", rule.UID, "org", rule.OrgID)
-		// 0 is passed as userID as then the quota logic will only check for
-		// the organization quota, as we don't have any user scope here.
-		_, err = prov.ruleService.CreateAlertRule(ctx, rule, alert_models.ProvenanceFile, 0)
+		// a nil user is passed in as then the quota logic will only check for
+		// the organization quota since we don't have any user scope here.
+		_, err = prov.ruleService.CreateAlertRule(ctx, nil, rule, alert_models.ProvenanceFile)
```

Comment:
let's create a background user with all the permissions needed: folders, data sources alert rules etc.

Reply:
Right now we have no way to ensure the UserID of a background user is `0`, so if that changes this would break. I want to keep the semantics of this as close to before as possible in this PR, probably worth looking at how to solve that in a follow up though.